src/test/osd/osdcap.cc

```diff
@@ -270,6 +270,11 @@ TEST(OSDCap, BasicRW) {
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", false, true, false, true));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, false, true, false));
   ASSERT_FALSE(cap.is_capable("bar", "", 0, "foo", true, true, true, true));
+
+  // class whitelist flag works
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", false, true, false, true, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, false, true, false, true));
+  ASSERT_TRUE(cap.is_capable("bar", "", 0, "foo", true, true, true, true, true));
```

Comment:
This doesn't look right... did you try runing this unit test?

The "allow rw" behavior shouldn't change.  But the "allow x" case should change.. where existing test cases where whitelisting is false should go from true -> false, and new cases should only be true if whitelist is true...